### PR TITLE
feat: add service handling user event when get in/out room

### PR DESCRIPTION
**Pull request by TMNhat1810**

### CODE REVIEW

File: app/socket/socket.js [hunk 1]
```diff
@@ -13,6 +14,7 @@ export const attachIO = (server) => {
   io.on(SOCKET_EVENT.CONNECTION, (socket) => {
     socket.on(SOCKET_EVENT.USER.JOIN, (data) => {
       const { userId, roomId } = data;
+      roomService.addUserToRoom(userId, roomId);
```
Comment (TMNhat1810) [inline app/socket/socket.js:17]: I dont put await here because I want it to process synchronously (tested, work ideally)

**Review by LongTCH — APPROVED** [review]

**Review by HoaiAnhTran — APPROVED** [review]

File: app/services/room.js [hunk 2]
```diff
@@ -61,4 +61,43 @@ export const roomService = {
...
+        const userIndex = room.voting.findIndex(
+          (userVoting) => userVoting.user === userId
+        );
+        if (userIndex !== -1) {
+          room.voting.splice(userIndex, 1);
+          await room.save();
+        }
+      }
+    } catch {
+      return false;
```
Comment (ces-hoangdao) [inline app/services/room.js:100]: why return false here?

Comment (TMNhat1810) [reply app/services/room.js:100]: because i want to inform that user is failed to remove

**Review by ces-hoangdao — APPROVED** [review]
Look good to me

**Review by ces-anduynguyen — APPROVED** [review]

**Pull request merged**